src/Employer/Employer.Web/Controllers/VacancyPreviewController.cs

```diff
@@ -70,6 +71,59 @@ public async Task<IActionResult> Submit(SubmitEditModel m)
             return View(ViewNames.VacancyPreview, viewModel);
         }
 
+        [HttpGet("approve-advert", Name = RouteNames.ApproveJobAdvert_Get)]
+        public IActionResult ApproveJobAdvert(VacancyRouteModel vm)
+        {              
+            return View(new ApproveJobAdvertViewModel());
+        }
+
+        [HttpPost("approve-advert", Name = RouteNames.ApproveJobAdvert_Post)]
+        public async Task<IActionResult> ApproveJobAdvert(ApproveJobAdvertViewModel vm)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View("ApproveJobAdvert");
+            }
+
+            if ((bool)vm.ApproveJobAdvert)
```

Comment:
why the cast rather than just `vm.ApproveJobAdvert` or `vm.ApproveJobAdvert.GetValueOrDefault()`

Reply:
done